File: consumer.js
const AWS = require('aws-sdk');
const sns = new AWS.SNS();

exports.handler = async (event) => {
  for (const record of event.Records) {
    const body = JSON.parse(record.body);

    await sns.publish({
      TopicArn: process.env.SNS_TOPIC_ARN,
      Subject: 'Notification from Consumer',
      Message: `Received message: ${body.message}`
    }).promise();
  }

  return { status: 'success' };
};

File: consumer.py
import os
import json
import boto3

sns = boto3.client('sns')

def handler(event, context):
    for record in event['Records']:
        body = json.loads(record['body'])

        sns.publish(
            TopicArn=os.environ['SNS_TOPIC_ARN'],
            Subject='Notification from Consumer',
            Message=f"Received message: {body['message']}"
        )

    return { 'status': 'success' }
